### PR TITLE
Add validations and create model specs

### PR DESCRIPTION
**Pull request by ivangonzalez224**
## Add validations and create model specs
I added the following features:

- Added validations for the name and the post_counter in the User model.
- Added validations for the title, comments_counter and likes_counter in the Post model.
- Created unit specs for the validations and methods for the User model.
- Created unit specs for the validations and methods for the Post model.
- Created unit specs for the validations and methods for the Comment model.
- Created unit specs for the validations and methods for the Like model.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      user = User.create(name: 'Ted', posts_counter: 0)
+      post = Post.new(user: user, title: 'Test Post', comments_counter: 0, likes_counter: 0)
+
+      expect do
+        post.save
+        user.reload
+      end.to change(user, :posts_counter).by(1)
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline spec/models/post_spec.rb]: - [ ] Well done for testing the `comments_counter`  and `likes_counter` so far 👏 Kindly make sure to test the following validations as we need add unit tests for all validations as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.) ⛏✨
  -  **_`likes_counter` must be greater than or equal to zero_**
  - **_`comments_counter` must be an integer_**

- [ ] Also, as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), we need to add tests for all model's methods, therefore, kindly make sure to test the `five_most_recent_comments` method present in `post.rb` 💪

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @ivangonzalez224 

Good job so far 👏! I really appreciate the effort you put in!  **_However, there are few things that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Validations are added
- ✅ Unit tests are present
- ✅ Descriptive PR and README file
- ✅ All implemented tests are passing

image


## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (ivangonzalez224) [reply spec/models/post_spec.rb]: Hi @KanzaTahreem thank you for the review.
Changes done: added tests for the likes_counter and comments_counter attributes, and for the five_most_recent_comments method.

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @ivangonzalez224 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Implemented suggested changes
-  ✅ Great work using the right Gitflow 
-  ✅ Your README looks good
-  ✅ All implemented tests are passing

image


## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**